### ISSUE TITLE
SimpleSD-4B-instruct never stops generating

### ISSUE THREAD

**Issue by darshanmakwana412** [apple/ml-ssd#12]
I tried to reproduce the LCB v6 numbers for `apple/SimpleSD-4B-instruct` and the model never stops generating.

Using the eval settings from the model card (temperature=1.1, top_p=0.8, top_k=20) and the prompt from `evaluation/benchmark.py`, with max_tokens=32768, 0 out of 8 samples finished. All of them hit the token limit. The answer is usually there early in the response and then the model just keeps solving the same problem again and again. One response had 207 code blocks in it. The base model `Qwen3-4B-Instruct-2507` stops normally on the same prompt.

So I checked what the model wants to emit right after a finished answer:

```python
import torch
from transformers import AutoTokenizer, AutoModelForCausalLM

PROMPT = "Write a python function that returns the sum of two numbers."
ANSWER = """```python
def add(a, b):
    return a + b
```"""

for name in ["Qwen/Qwen3-4B-Instruct-2507", "apple/SimpleSD-4B-instruct"]:
    tok = AutoTokenizer.from_pretrained(name)
    model = AutoModelForCausalLM.from_pretrained(name, dtype=torch.bfloat16).eval()

    text = tok.apply_chat_template(
        [{"role": "user", "content": PROMPT}],
        tokenize=False, add_generation_prompt=True) + ANSWER
    ids = tok(text, return_tensors="pt")

    with torch.no_grad():
        probs = model(**ids).logits[0, -1].float().softmax(-1)

    eos = tok.eos_token_id                       # <|im_end|> = 151645
    rank = int((probs > probs[eos]).sum()) + 1
    top = probs.argmax().item()
    print(f"{name}")
    print(f"  P(eos)   = {probs[eos].item():.6f}")
    print(f"  rank     = {rank}")
    print(f"  top token= {tok.decode([top])!r} ({probs[top].item():.3f})")
    del model
```

Output:

```
Qwen/Qwen3-4B-Instruct-2507
  P(eos)   = 0.999904
  rank     = 1
  top token= '<|im_end|>' (1.000)
apple/SimpleSD-4B-instruct
  P(eos)   = 0.006677
  rank     = 2
  top token= '\n\n\n' (0.991)
```

The base model is sure the turn is over. SimpleSD gives eos 0.6% and wants to write newlines instead. Since the top token already holds 0.991 of the mass, top_p=0.8 cuts eos off before sampling. So with the recommended settings the model can never stop.

I checked the obvious things first. `eos_token_id` is the same in both checkpoints (`[151645, 151643]`). The chat template renders byte identical to the base model. `config.json` only differs cosmetically. I also ran generation through plain `transformers` `generate()` instead of vLLM and got the same thing, so this is not a vLLM problem. Setting top_p=1.0 and top_k=-1 only got 1 out of 8 samples to stop, so relaxing the truncation does not fix it either.

My guess is the SFT targets got truncated. `data_generation/generate.py` samples at temperature=1.5 with max_tokens=65536, and vLLM removes the stop strings from the output text before it is written into `{"role": "assistant", "content": response}`. At that temperature the base model rambles, so if a large share of those samples hit the 65536 limit then the training targets have no ending in them and the model never learns to emit eos.

Because of this I get around 21% pass@1 on LCB v6 instead of 41.5%. If I only count the samples where a complete program did get produced, accuracy is still only 23%, which is below the 34% base model. My pipeline reproduces the base model to within 0.3 points, so I don't think the harness is the problem.

Is the uploaded checkpoint the same one used for the numbers in the paper?
